service/src/main/java/bio/terra/externalcreds/controllers/OidcApiController.java

```diff
@@ -25,32 +25,15 @@
 import org.springframework.stereotype.Controller;
 
 @Controller
-public class OidcApiController implements OidcApi {
-
-  private final HttpServletRequest request;
-  private final LinkedAccountService linkedAccountService;
-  private final ObjectMapper mapper;
-  private final ProviderService providerService;
-  private final SamService samService;
-  private final PassportService passportService;
-  private final AuditLogger auditLogger;
-
-  public OidcApiController(
-      HttpServletRequest request,
-      LinkedAccountService linkedAccountService,
-      ObjectMapper mapper,
-      PassportService passportService,
-      ProviderService providerService,
-      SamService samService,
-      AuditLogger auditLogger) {
-    this.request = request;
-    this.linkedAccountService = linkedAccountService;
-    this.mapper = mapper;
-    this.passportService = passportService;
-    this.providerService = providerService;
-    this.samService = samService;
-    this.auditLogger = auditLogger;
-  }
+public record OidcApiController(
```

Comment:
nifty

Reply:
Of note: records are implicitly final; this doesn't work for anything deeper than a controller, like a service. Also, intellij doesn't quite understand them perfectly yet, so it thinks they're unused, but compilation has no errors.